packages/front-end/pages/sdks/webhooks.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from "react";
+import { WebhookInterface } from "back-end/types/webhook";
+import {
+  FaCheck,
+  FaInfoCircle,
+  FaPencilAlt,
+  FaPlusCircle,
+} from "react-icons/fa";
+import { ago } from "shared/dates";
+import useApi from "@/hooks/useApi";
+import WebhooksModal from "@/components/Settings/WebhooksModal";
+import DeleteButton from "@/components/DeleteButton/DeleteButton";
+import { useAuth } from "@/services/auth";
+import Tooltip from "@/components/Tooltip/Tooltip";
+
+export default function SDKWebhooks({ sdkid }) {
+  const { data, mutate } = useApi(`/webhooks/sdk/${sdkid}`);
+  const [
+    createModalOpen,
+    setCreateModalOpen,
+  ] = useState<null | Partial<WebhookInterface>>(null);
+  const { apiCall } = useAuth();
+
+  const renderTableRows = () => {
+    console.log(data?.webhooks);
+
+    // only rener table is there is data to show
+    return data?.webhooks?.map((webhook) => (
+      <tr key={webhook.name}>
+        <td>{webhook.name}</td>
+        <td>{webhook.endpoint}</td>
+        <td>{webhook.sendPayload === true ? "yes" : "no"}</td>
+        <td>{webhook.signingKey}</td>
+        <td>
+          {webhook.error ? (
+            <pre className="text-danger">Error</pre>
+          ) : webhook.lastSuccess ? (
+            <em>
+              <FaCheck className="text-success" /> last fired{" "}
+              {ago(webhook.lastSuccess)}
+            </em>
+          ) : (
+            <em>never fired</em>
+          )}
+        </td>
+        <td>
+          <a
+            href="#"
+            className="tr-hover text-primary mr-3"
+            title="Edit this webhook"
+            onClick={(e) => {
+              e.preventDefault();
+              setCreateModalOpen(webhook);
+            }}
+          >
+            <FaPencilAlt />
+          </a>
+          <DeleteButton
+            link={true}
+            className={"tr-hover text-primary"}
+            displayName="Webhook"
+            title="Delete this webhook"
+            onClick={async () => {
+              await apiCall(`/webhook/${webhook.id}`, {
+                method: "DELETE",
+              });
+              mutate();
+            }}
+          />
+        </td>
+      </tr>
+    ));
+  };
+  const renderEmptyState = () => (
+    <>
+      <button
```

Comment:
should add (and other edit buttons) be removed when permissions aren't set?

Reply:
what is the permissions for sdks?